### PR TITLE
Feature/mams 4 adding new archive category and getting archive category

### PR DESCRIPTION
**Pull request by KatBre**

### CODE REVIEW

**Review by MichalKulygin — APPROVED** [review]

File: src/main/java/pl/sdaproject/medicalarchivemanagementsystem/dto/ArchiveCategoryRequest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.Getter;
+import org.hibernate.validator.constraints.Length;
+
+import javax.validation.constraints.NotBlank;
+
+@Getter
+@Builder
+public class ArchiveCategoryRequest {
+    @NotBlank
+    @Length(min = 1)
```
Comment (klezner) [inline src/main/java/pl/sdaproject/medicalarchivemanagementsystem/dto/ArchiveCategoryRequest.java]: Add @Pattern and regexp to specify type of archiveCategory

File: src/main/java/pl/sdaproject/medicalarchivemanagementsystem/dto/ArchiveCategoryRequest.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    private String archiveCategory;
```
Comment (klezner) [inline src/main/java/pl/sdaproject/medicalarchivemanagementsystem/dto/ArchiveCategoryRequest.java]: Consider using different name of variable - same as class name

File: src/main/java/pl/sdaproject/medicalarchivemanagementsystem/dto/ArchiveCategoryRequest.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    @NotBlank
+    @Length(min = 1)
```
Comment (klezner) [inline src/main/java/pl/sdaproject/medicalarchivemanagementsystem/dto/ArchiveCategoryRequest.java]: Use @Min - value

File: src/main/java/pl/sdaproject/medicalarchivemanagementsystem/dto/ArchiveCategoryResponse.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package pl.sdaproject.medicalarchivemanagementsystem.dto;
+
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+@Builder
+public class ArchiveCategoryResponse {
+    private Long id;
+    private String archiveCategory;
```
Comment (klezner) [inline src/main/java/pl/sdaproject/medicalarchivemanagementsystem/dto/ArchiveCategoryResponse.java]: Variable name, maybe differenet name?

File: src/main/java/pl/sdaproject/medicalarchivemanagementsystem/mapper/ArchiveCategoryMapper.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import pl.sdaproject.medicalarchivemanagementsystem.dto.ArchiveCategoryResponse;
+import pl.sdaproject.medicalarchivemanagementsystem.model.ArchiveCategory;
+
+@Component
+public class ArchiveCategoryMapper {
+
+    public ArchiveCategoryResponse mapArchiveCategoryToArchiveCategoryResponse(ArchiveCategory archiveCategory) {
+
+        return ArchiveCategoryResponse.builder()
+                .archiveCategory(archiveCategory.getArchiveCategory())
```
Comment (klezner) [inline src/main/java/pl/sdaproject/medicalarchivemanagementsystem/mapper/ArchiveCategoryMapper.java]: As above

File: src/main/java/pl/sdaproject/medicalarchivemanagementsystem/model/ArchiveCategory.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Getter
+public class ArchiveCategory {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NotBlank
+    @Length(min = 1)
```
Comment (klezner) [inline src/main/java/pl/sdaproject/medicalarchivemanagementsystem/model/ArchiveCategory.java]: Use @Pattern

File: src/main/java/pl/sdaproject/medicalarchivemanagementsystem/model/ArchiveCategory.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    private String archiveCategory;
+    @NotBlank
+    @Length(min = 1)
```
Comment (klezner) [inline src/main/java/pl/sdaproject/medicalarchivemanagementsystem/model/ArchiveCategory.java]: Use @Min

File: src/main/java/pl/sdaproject/medicalarchivemanagementsystem/model/ArchiveCategory.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+@NoArgsConstructor
+@Builder
+@Getter
+public class ArchiveCategory {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NotBlank
+    @Length(min = 1)
+    private String archiveCategory;
```
Comment (klezner) [inline src/main/java/pl/sdaproject/medicalarchivemanagementsystem/model/ArchiveCategory.java]: Maybe different variable name

File: src/main/java/pl/sdaproject/medicalarchivemanagementsystem/service/ArchiveCategoryService.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class ArchiveCategoryService {
+
+    private final ArchiveCategoryRepository archiveCategoryRepository;
+
+    public ArchiveCategory createArchiveCategory(String category, Integer years) {
+        final ArchiveCategory archiveCategory = ArchiveCategory.builder()
+                .archiveCategory(category)
```
Comment (klezner) [inline src/main/java/pl/sdaproject/medicalarchivemanagementsystem/service/ArchiveCategoryService.java]: As above

File: src/main/resources/application.properties [hunk 10]
```diff
@@ -1 +1,3 @@
-
+spring.jpa.hibernate.ddl-auto=update
+spring.datasource.url=jdbc:h2:file:./mams;AUTO_SERVER=TRUE;
+spring.jpa.show-sql=true
```
Comment (klezner) [inline src/main/resources/application.properties]: No commit for this file :)

**Review by MichalKulygin — APPROVED** [review]

File: src/main/java/pl/sdaproject/medicalarchivemanagementsystem/dto/ArchiveCategoryRequest.java [hunk 11]
```diff
@@ -2,17 +2,19 @@
...
 
 @Getter
 @Builder
 public class ArchiveCategoryRequest {
     @NotBlank
-    @Length(min = 1)
-    private String archiveCategory;
-    @NotBlank
-    @Length(min = 1)
+    @Pattern(regexp = "[A-Z]+")
```
Comment (klezner) [inline src/main/java/pl/sdaproject/medicalarchivemanagementsystem/dto/ArchiveCategoryRequest.java:15]: "[A-Z][0-9]{2}"??

File: src/main/resources/application.properties [hunk 12]
```diff
@@ -1,3 +1,3 @@
 spring.jpa.hibernate.ddl-auto=update
 spring.datasource.url=jdbc:h2:file:./mams;AUTO_SERVER=TRUE;
-spring.jpa.show-sql=true
+spring.jpa.show-sql=true
```
Comment (klezner) [inline src/main/resources/application.properties:3]: ?

**Review by klezner — APPROVED** [review]

**Pull request merged**